### ISSUE TITLE
Avarage_F1 = 0.000 After 8 epochs

### ISSUE THREAD

**Issue by wuhang03** [SunnyHaze/IML-ViT#32]
Dear author, thank you for your brilliant work and codes here. When I try to reproduce the results and train the model, I set the parameters exactly as the .sh in your repository, but after 8 epochs, the f1 score calculated is 0.0000. I wonder if this is normal, and what problems may cause this. Loss seems to be okay and the training is smooth.
Appreciate it very much if you can reply.

[16:37:54.545643] Test: [8]  [229/230]  eta: 0:00:00  average_f1: 0.0000 (0.0167)  time: 0.1405  data: 0.0002  max mem: 18097
[16:37:54.621332] Test: [8] Total time: 0:00:33 (0.1451 s / it)
[16:37:55.029155] Averaged stats: average_f1: 0.0000 (0.0167)

**Comment by SunnyHaze** [comment]
Hi! Thanks for your attention to our project!
Please refer to issue #26 first. If the problem still exists, feel free to reach out.

Thank you.

**Comment by wuhang03** [comment]
OK got it !!
I will wait for some more epochs and see the result.
BTW, your response is really fast hhh.
Thank you very much !!!

**Comment by SunnyHaze** [comment]
I hope it will work. It's my pleasure.

**Comment by wuhang03** [comment]
{"train_lr": 9.900525287686401e-05, "train_predict_loss": 0.12908920126740614, "train_edge_loss": 0.042181301665729226, "test_average_f1": 0.017881513795647928, "epoch": 16}
{"train_lr": 9.827102807611127e-05, "train_predict_loss": 0.10702291691243074, "train_edge_loss": 0.04218290319320678, "test_average_f1": 0.08120784876364784, "epoch": 20}
{"train_lr": 9.733947457051792e-05, "train_predict_loss": 0.1030729257036406, "train_edge_loss": 0.04541666851110918, "test_average_f1": 0.0, "epoch": 24}

Did I set something wrong? After 24 epochs it is still 0.0.
The loss is decreasing, but the f1 score is really weird.

**Comment by SunnyHaze** [comment]
Please specify your number of GPU, batchsize, and accu_iters.

**Comment by wuhang03** [comment]
python main_train.py \
  --world_size 1 \
  --batch_size 2 \
  --data_path "/root/autodl-tmp/CASIA2.0" \
  --epochs 200 \
  --lr 1e-4 \
  --min_lr 5e-7 \
  --weight_decay 0.05 \
  --edge_lambda 20 \
  --predict_head_norm "BN" \
  --vit_pretrain_path "./pretrained-weights/mae_pretrain_vit_base.pth" \
  --test_data_path "/root/autodl-tmp/CASIA1.0" \
  --warmup_epochs 4 \
  --output_dir ./output_dir/ \
  --log_dir ./output_dir/  \
  --accum_iter 32 \
  --seed 42 \
  --test_period 4 \
  --num_workers 8 \
  2> train_error.log 1>train_log.log

I don't use distributed training since I only get one 4090.
  
  I doubt if this is kind of dataset problem, since the loss keeps decreasing, indicating that the training is working.
Maybe there is something wrong with CASIA1.0
In CASIA 1.0, there are Modified Tp.zip and Original Tp.zip, which one should I use

**Comment by SunnyHaze** [comment]
Change accum_iter to 8 may perform better.

I am not sure about the dataset issue. Plz use Tensorboad to visualize and check if the output is correct.

**Comment by SunnyHaze** [comment]
This is our latest experiment on parameter finetuning. Honestly, from my personal experience, accum_iter is an important parameter and should not be very large. I will consider modify this in current repo.
![image](https://github.com/user-attachments/assets/79afec87-8a5a-4713-9d2f-d70fba919900)

**Comment by wuhang03** [comment]
Your suggestion is really useful.
I change the accum_iter and it is trained pretty smoothly.
Moreover, I transfer your model to other tasks and datasets, and it performs perfect !
Thank you again for your brilliant work and codes.

**Comment by SunnyHaze** [comment]
That's great! I will revise the default settings to accum_iter to 8 today.
If you find our code helpful, recommending it to your friends or co-workers will be the greatest encouragement to us.

**Comment by SunnyHaze** [comment]
default value for --accum_iter is solved with this commit: https://github.com/SunnyHaze/IML-ViT/commit/006f410618c542fdebcaa380ee0c5359d77a57cc
